app/src/main/java/fi/bitrite/android/ws/ui/MapFragment.java

```diff
@@ -123,36 +122,25 @@
     private static final int POSITION_PRIORITY_LAST_STORED = 2;
     private static final int POSITION_PRIORITY_FORCED = 100;
 
-    private int mLastPositionType = -1;
-    private CameraPosition mLastCameraPosition = null;
-    private FusedLocationProviderClient mFusedLocationClient;
-    private Location mLastDeviceLocation;
-    private final LocationCallback mLocationCallback = new LocationCallback() {
-        @Override
-        public void onLocationResult(LocationResult locationResult) {
-            if (locationResult == null) {
-                return;
-            }
+    private int mLastPositionType;
+    private ZoomedLocation mLastPosition;
+    private ZoomedLocation mOsmdroidBug1055_positionBeingSet;
```

Comment:
https://github.com/osmdroid/osmdroid/commit/c423668d105b56d466944ea8daf472e27836903f#diff-693ee9be95e9fc9b05e9f012479dc3ab

This proposed fix for https://github.com/osmdroid/osmdroid/issues/1055 seems to have found its way into the master branch and should have been released in the newest 6.0.2 version.

Could you test that and, if it works, remove your workarounds?

Reply:
The bug is no longer around but other issues that prevent removing this completely. A cleanup is done in febf1da.